api/src/main/java/org/eclipse/microprofile/config/Config.java

```diff
@@ -69,13 +69,11 @@
      *
      * @param propertyName
      *             The configuration propertyName.
-     * @return The resolved property value as a String-Optional, which will bypass converters.
+     * @return The resolved property value as a String, which will bypass any converters.
      *
-     * @throws IllegalArgumentException
-     *             is thrown if the propertyName maps to an object that is not a
-     *             String.
+     * 
      */
-    Optional<String> getString(String propertyName);
+    String getRawString(String propertyName);
```

Comment:
`Optional` as mentioned in recent calls is quite ugly especially when you try to combine it with type-safety you quickly stretch it beyond its limit. 
Typesafe Config just to give one example has `String getString(String path);` and getters for most other base types including primitives. `ConfigObject getObject(String path);` returns a `ConfigObject` which is a specialized `ConfigValue` And `Object getAnyRef(String path);` is the "raw" equivalent simply returning the underlying value as `Object`. No `Optional`, although the library is now strongly tailored to Java SE 8, e.g. HOCON dealing with java.time values like `LocalDate` or `Duration`.